src/database/DataTypes/RequisitionItem.js

```diff
@@ -93,11 +93,10 @@ export class RequisitionItem extends Realm.Object {
    * @return  {TransactionItem}
    */
   get linkedTransactionItem() {
-    // TODO: RequisitionItem.Requisition is optional field, unhandled possible TypeError.
-    // TODO: isRequest is Requisition field, will always return undefined.
-    if (this.isRequest || !this.requisition.linkedTransaction) return null;
+    if (this.requisition.isRequest || !this.requisition.linkedTransaction) {
+      return null;
+    }
 
-    // TODO: Requisition.linkedTransaction is optional field, unhandled possible TypeError.
     return this.requisition.linkedTransaction.items.filtered('item.id == $0', this.item.id)[0];
```

Comment:
Think this can be better written (more consistent with existing codebase which uses boolean short-circuits for this stuff).

```
if (this.requisition.isRequest) return null;
return this.requisition.linkedTransaction && this.requisition.linkedTransaction.items.filtered('item.id == $0', this.item.id)[0];
```

Reply:
This would change the behaviour a little bit, if `this.requisition.linkedTransaction` is falsy then the method will return `false` rather than `null`. I'd rather keep it returning null

Reply:
I think? I want to test this now.

Reply:
I'm wrong, though if x is `undefined`, it'll return that instead of `null` which still is a bit inconsistent.

Reply:
Shouldn't be an issue, optional fields in Realm should be null if not defined otherwise.